Content.Client/Light/Visualizers/EmergencyLightVisualizerSystem.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Content.Shared.Light.Component;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Light.Visualizers;
+
+public sealed class EmergencyLightVisualizerSystem : VisualizerSystem<EmergencyLightVisualizerComponent>
+{
+    protected override void OnAppearanceChange(EntityUid uid, EmergencyLightVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+
+        if (!AppearanceSystem.TryGetData(uid, EmergencyLightVisuals.On, out bool on, args.Component))
+            on = false;
+
+        args.Sprite.LayerSetState(EmergencyLightVisualLayers.Light, on ? "emergency_light_on" : "emergency_light_off");
+        args.Sprite.LayerSetShader(EmergencyLightVisualLayers.Light, on ? "unshaded" : "shaded");
```

Comment:
unhardcode this, one way i can think of is by mapping two layers (one visible and other invisible) with the required state and shader, then get it here and swap visibility as needed. Tho if you find a simpler solution do that